Data/Hashable.hs

```diff
@@ -231,13 +239,36 @@ instance Eq a => Eq (Hashed a) where
 instance Ord a => Ord (Hashed a) where
   Hashed a _ `compare` Hashed b _ = a `compare` b
 
-instance Hashable a => Hashable (Hashed a) where
+instance Show a => Show (Hashed a) where
+  showsPrec d (Hashed a _) = showParen (d > 10) $
+    showString "hashed" . showChar ' ' . showsPrec 11 a
+
+instance Hashable (Hashed a) where
   hashWithSalt = defaultHashWithSalt
   hash (Hashed _ h) = h
 
+-- This instance is a little unsettling. It is unusal for
+-- 'liftHashWithSalt' to ignore its first argument when a
+-- value is actually available for it to work on.
+instance Hashable1 Hashed where
+  liftHashWithSalt _ s (Hashed _ h) = defaultHashWithSalt s h
```

Comment:
Since `Hashed` can only be created by `hashed`, that would require introducing another                                                                   function `hashedWithSalt` and changing `Hashed` to:

    data Hashed = Hashed a (Maybe Int)

That being said, I don't like that because it's easily to accidentally use incorrectly, circumventing the caching its supposed to provide. My preference would be to leave it as is and document the counterintuitive behavior of `hashWithSalt` on a `Hashed`. For the libraries I'm working on, caching is the most important thing that `Hashed` provides.